homeassistant/components/growatt_server/config_flow.py

```diff
@@ -0,0 +1,84 @@
+"""Config flow for growatt server integration."""
+import growattServer
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+
+from .const import CONF_PLANT_ID, DOMAIN  # noqa # pylint: disable=unused-import
+
+
+class GrowattServerConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow class."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialise growatt server flow."""
+        self.user_input = {}
+        self.api = growattServer.GrowattApi()
+        self.plants = {}
```

Comment:
We don't need to define `self.plants` as an instance attribute if we inline the plant form in the plant step.